cmd/gossamer/configcmd.go

```diff
@@ -143,4 +156,60 @@ func startP2PService(cfg *p2p.ServiceConfig) *p2p.Service {
 		log.Error("error starting p2p", "err", err.Error())
 	}
 	return srv
+}
+
+// dumpConfig is the dumpconfig command.
+func dumpConfig(ctx *cli.Context) error {
+	fig, err := setConfig(ctx)
+	if err != nil {
+		return err
+	}
+	comment := ""
+
+	out, err := tomlSettings.Marshal(&fig)
```

Comment:
It seems setConfig will return nil if the file exists but is empty, causing this to fail. Should probably check somewhere in `loadConfig` or `setConfig`